gammapy/maps/geom.py

```diff
@@ -629,6 +629,10 @@ def assert_names(self, required_names):
         try:
             for ax, required_name in zip(self, required_names):
                 ax.assert_name(required_name)
+
+            if not len(self) == len(required_names):
```

Comment:
Shouldn't this be checked before the name assert?

Reply:
Yes, done.